addon/components/ui-radio.js

```diff
@@ -1,24 +1,82 @@
 import Ember from 'ember';
-import CheckboxMixin from '../mixins/checkbox-mixin';
+import Checkbox from '../mixins/checkbox';
+import { isPromise } from '../utils/promise-tools';
+import { isFulfilled } from '../utils/promise-tools';
+import { getPromiseContent } from '../utils/promise-tools';
+import PromiseResolver from '../mixins/promise-resolver';
 
-export default Ember.Component.extend(CheckboxMixin, {
+export default Ember.Component.extend(Checkbox, PromiseResolver, {
   type: 'radio',
-  classNames: [ 'radio' ],
-  name: 'default',
+  classNames: ['radio'],
+  ignorableAttrs: ['checked', 'label', 'disabled', 'value', 'current'],
 
-  init: function() {
+  init() {
     this._super(...arguments);
 
-    if (!(this.get('name') && this.get('name') !== 'default')) {
-      Ember.Logger.warn('Name was not passed into semantic radio component');
+    if (Ember.isBlank(this.get('name'))) {
+      this.set('name', 'default');
+      Ember.Logger.warn("The required component parameter of 'name' was not passed into the ui-radio component");
     }
   },
 
-  _onChange: function() {
-    this.set('current', this.get('value'));
+  // Internal wrapper for onchange, to pass through checked
+  _onChange() {
+    let value = this.get('value');
+    return this.attrs.onChange(value, this);
   },
 
-  checked: Ember.computed('current', 'value', function() {
-    return this.get('current') === this.get('value');
-  })
+  didInitSemantic() {
+    this._super(...arguments);
+    this._inspectValueAndCurrent();
+  },
+
+  didUpdateAttrs() {
+    this._super(...arguments);
+    this._inspectValueAndCurrent();
+  },
+
+  _inspectValueAndCurrent() {
+    let value = this.get('value');
+    let current = this.get('current');
+    // If either are a promise, we need to make sure both are resolved
+    // Or wait for them to resolve
+    if (isPromise(value) || isPromise(current)) {
+
+      // This code is probably overkill, but i wanted to ensure that
+      // if the promises are resolved we render as soon as possible instead of waiting
+      // for the hash to resolve each time
+      if (isPromise(value)) {
+        if (!isFulfilled(value)) {
+          return this.resolvePromise(Ember.RSVP.hash({ value, current }), this._checkValueAndCurrent);
+        } else {
+          value = getPromiseContent(value);
+        }
+      }
+
+      if (isPromise(current)) {
```

Comment:
The biggest reason I didn't was because on I needed to return early if it is a promise. Felt like I would be adding extra code to see if the promise returned early. Do you have an idea for a better internal function signature?